presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -76,25 +79,27 @@ public PlanNode optimize(PlanNode plan, Session session, Map<Symbol, Type> types
 
         Memo memo = new Memo(idAllocator, plan);
         Lookup lookup = Lookup.from(memo::resolve);
+        Matcher matcher = DEFAULT_MATCHER;
```

Comment:
a different Matcher implementation is needed for resolving GroupReferences when navigating via properites (`.with(source().matching(...))`).

Extracted the `Parameterize IterativeOptimizer methods with Matcher` commit just before switching to the `PlanNodeMatcher` implementation (which is needed for matching nested nodes)